pkg/cluster/nodes/node.go

```diff
@@ -259,6 +263,24 @@ func (n *Node) Ports(containerPort int) (hostPort int, err error) {
 	return n.nodeCache.ports[containerPort], nil
 }
 
+// Role returns the role of the node
+func (n *Node) Role() (role config.NodeRole, err error) {
+	// use the cached version first
+	if n.nodeCache.role != "" {
+		return n.nodeCache.role, nil
+	}
+	// retrive the role the node using docker inspect
+	lines, err := docker.Inspect(n.nameOrID, fmt.Sprintf("{{index .Config.Labels %q}}", constants.ClusterRoleKey))
```

Comment:
We should probably move logic to obtain labels into the docker package at some point. This is consistent for now though and this PR is nice and small 👍

Reply:
:+1: